clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManager.java

```diff
@@ -161,4 +161,28 @@ protected Checkpoint createCheckpoint(RocksDB rocksDB) {
     return Checkpoint.create(rocksDB);
   }
 
+  /**
+   * util method to create a snapshot for batch only
+   */
+  public static void createSnapshotForBatch(RocksDB rocksDB, String fullPathForPartitionDBSnapshot) {
```

Comment:
+1 what jialin suggested. It's great to make `BlobSnapshotManager` a central class to manage rocksdb blob creation so if we can make it generic w/o significant change, let's do so.

> Also, why it is not acceptable when the path is empty? I think empty indicates there is nothing there and we can create stuff there.

This is copied from the original logic i believe. The original logic would create snapshot directory first before creating the snapshot. If the directory isn't present, it means either the `blobTransferEnabled` feature is not enabled or there's some disk issue hence skipping the snapshotting.

Reply:
`do you think you want to limit the scope first and then expand to hybrid? ` Yes, we will extend the scope to hybrid later. Will rename it to `createSnapshot` once it can be leveraged by hybrid mode. But for distinguishing, lets name it createSnapshotForBatch for now. 

`why it is not acceptable when the path is empty? should we delete everything if the folder is not clean? ` This logic is copy from the original logic. I think previously it checks the path string is a empty string. (@adamxchen Seem it is not check the directory presence, fullPathForPartitionDBSnapshot here is a string) It is a good point that we should clean old snapshot.